### ISSUE TITLE
[deploy-sourcegraph] Write 2.x. -> 3.0 migraiton docs

### ISSUE THREAD

**Issue by ggilmore** [sourcegraph/sourcegraph#1692]
Depends on: https://github.com/sourcegraph/sourcegraph/issues/1682
Depends on: https://github.com/sourcegraph/sourcegraph/issues/1428

After the above changes are made to https://github.com/sourcegraph/deploy-sourcegraph, https://github.com/sourcegraph/deploy-sourcegraph/blob/master/docs/migrate.md will need to be updated with fully-tested instructions for sourcegraph 3.0

Also @chrismwendt, https://github.com/sourcegraph/deploy-sourcegraph/blob/master/docs/migrate.md should be updated with explanations about the changes to language servers (i.e. extensions, removal of lsp-proxy and langserver deployments, etc.) 

- [ ] Update https://github.com/sourcegraph/deploy-sourcegraph/blob/master/docs/migrate.md with tested changes related to the management console @slimsag 
- [ ] Update https://github.com/sourcegraph/deploy-sourcegraph/blob/master/docs/migrate.md with tested changes related to ingress-nginx @keegancsmith 
- [ ] Update https://github.com/sourcegraph/deploy-sourcegraph/blob/master/docs/migrate.md with tested changes related to extensions @chrismwendt 

_You can use https://github.com/sourcegraph/deploy-sourcegraph/blob/master/README.dev.md for instructions on how to spin up a cluster to test your migration documentation on_.

**Comment by nicksnyder** [comment]
The current migrate.md is really old and not applicable to 3.0. @ggilmore can you create a separate migration doc specifically for 3.0?

For management console, what migration needs to happen? Removing old auth provider configs from configmap?

**Comment by nicksnyder** [comment]
@ggilmore can you split this into multiple issues? It looks like @slimsag already has a tracking issue for management console at https://github.com/sourcegraph/sourcegraph/issues/1428

**Comment by slimsag** [comment]
To be clear, management console migration is automatic as long as you still have the old config file mounted in your deployment (i.e. as long as you didn't get rid of the configmap). I just need to document it. #1428 is about doing this.

**Comment by ggilmore** [comment]
> @ggilmore can you split this into multiple issues? It looks like @slimsag already has a tracking issue for management console at #1428

Just saw this - splitting into separate issues now
